File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        for i,currentnum in enumerate(nums):
            needed_num = target-currentnum
            if needed_num in seen:
                return[seen[needed_num],i]
            seen[currentnum] = i
        return[]
        

File: Data Structures & Algorithms/contains-duplicate-ii/submission-0.ts
class Solution {
    containsNearbyDuplicate(nums: number[], k: number): boolean {
        let seen = new Map<number, number>()

        for (let i = 0; i < nums.length; i++) {
            if (seen.has(nums[i])) {
                let previousIndex = seen.get(nums[i])!
                if (i - previousIndex <= k) {
                    return true
                }
            }
            seen.set(nums[i], i)
        }

        return false
    }
}